src/qoi/src/spectroscopic_absorption.C

```diff
@@ -63,4 +63,12 @@ namespace GRINS
     QoIBase::_qoi_value = sys_qoi;
   }
 
+  void SpectroscopicAbsorption::finalize_derivative(libMesh::NumericVector<libMesh::Number> & derivatives)
+  {
+    if (!derivatives.closed())
+      derivatives.close();
+
+    derivatives.scale(-100.0 * QoIBase::_qoi_value);
```

Comment:
Found a problem here: apparently the adjoint residual error estimator doesn't compute the QoI value before doing the error estimate, meaning that `QoIBase::_qoi_value == 0` and we converge without doing any AMR.

I can reorder the call sequence in `MeshAdaptiveSolver` to calculate the QoI value before doing the error estimate, but it feels a bit hacky to me.

I'm unsure of the proper way to modify the AMR code to account for when the QoI error estimate depends explicitly on the QoI value.